### PR TITLE
Allow parens inside arguments.

### PR DESCRIPTION
**Pull request by aquach**
I can't figure out how to write a test for it, but it seems to work on a few cases I manually tried. Should fix my CSSComb issue as well as #76.

### CODE REVIEW

**Comment by tonyganch** [comment]
Tests are organized in this way: `test/<syntax_name>/<node_type>`.
Every test has two files: `x.<syntax_name>` (a string we need to parse) and `x.json` (a tree that should be a result of parsing).
File `test.cofee` lists all tests which should be run.

For example, in your case:
- go to `test/less/arguments`;
- create files `5.less` and `5.json`;
- add this line to `test/less/arguments/test.coffee`: `it '5', -> this.shouldBeOk()`.

Take a look at files inside `test/less/arguments` as examples.

There are a few test helpers I find useful from time to time:
(1) `npm test less` will build sources and run tests only for less syntax.
(2) `npm run build && ./bin/gonzales -s less -c arguments ./test/less/arguments/5.less > ./test/less/arguments/5.json` will build sources and generate a test file with parsing tree for a given test. Format for using cli is: `./bin/gonzales -s <syntax_name> -c <node_type> <syntax_test_file> > <json_test_file>`.
(3) `npm run autofix-tests less` will build sources and generate test files with parsing trees _for all_ `*.less` files. You should use it with caution and only when you're working with a feature that changes a lot of cases and you are lazy enough to change all `*.json` files manually.

File: bin/gonzales.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ function processInputData(input) {
         process.stdout.write(ast.toJson());
         process.exit(0);
     } catch (e) {
-        if (!options.silent) process.stderr.write(e.message);
+        if (!options.silent) process.stderr.write(e.toString());
```
Comment (tonyganch) [inline bin/gonzales.js:50]: Could you please explain this change to me and move it to a separate commit?

Comment (aquach) [inline bin/gonzales.js:50]: Sure, sorry. It prints the ParseError stack message that shows the problematic line rather than just a terse error message, which made it a lot easier to debug.

**Comment by aquach** [comment]
Thanks so much for explaining! In this case, I tried to write a test like this:

In `5.less`:

`((2 / 3))`

I expect this to be parsable using the `-c arguments` flag, but for some reason it doesn't work: just outputs `{}`. I'm not sure what this means, especially since I don't have a stack trace. When I run gonzales without -c on an entire valid LESS file, it parses just fine:

```
html {
  width: calc((2 / 3));
}
```

Here's a snippet of what comes out:

```
                        {
                          "type": "arguments",
                          "content": [
                            {
                              "type": "parentheses",
                              "content": [
                                {
                                  "type": "number",
                                  "content": "2",
                                  "syntax": "less",
                                  "start": {
                                    "line": 2,
                                    "column": 16
                                  },
                                  "end": {
                                    "line": 2,
                                    "column": 16
                                  }
                                },
                                {
                                  "type": "space",
                                  "content": " ",
                                  "syntax": "less",
                                  "start": {
                                    "line": 2,
                                    "column": 17
                                  },
                                  "end": {
                                    "line": 2,
                                    "column": 17
                                  }
                                },
                                {
                                  "type": "operator",
                                  "content": "/",
                                  "syntax": "less",
                                  "start": {
                                    "line": 2,
                                    "column": 18
                                  },
                                  "end": {
                                    "line": 2,
                                    "column": 18
                                  }
                                },
                                {
                                  "type": "space",
                                  "content": " ",
                                  "syntax": "less",
                                  "start": {
                                    "line": 2,
                                    "column": 19
                                  },
                                  "end": {
                                    "line": 2,
                                    "column": 19
                                  }
                                },
                                {
                                  "type": "number",
                                  "content": "3",
                                  "syntax": "less",
                                  "start": {
                                    "line": 2,
                                    "column": 20
                                  },
                                  "end": {
                                    "line": 2,
                                    "column": 20
                                  }
                                }
                              ],
                              "syntax": "less",
                              "start": {
                                "line": 2,
                                "column": 15
                              },
                              "end": {
                                "line": 2,
                                "column": 21
                              }
                            }
                          ],
                          "syntax": "less",
                          "start": {
                            "line": 2,
                            "column": 14
                          },
                          "end": {
                            "line": 2,
                            "column": 22
                          }
                        }
```

Also, what's the difference between the `dev` and `master` branch? Which one should I be PR'ing?

**Comment by tonyganch** [comment]
TL;DR: You need to check for parentheses inside `checkArgument` function and get them inside `getArgument`.

For every node type there are two functions: check and get. `check` tests whether a substring looks like a given node type and `get` transforms the substring to a parsing tree. When you want to change some logic, you need to modify both function. In your commit you've modified code inside `getArguments` but left `checkArguments` untouched. That's the reason test is weird.

**Comment by tonyganch** [comment]
Here: https://github.com/tonyganch/gonzales-pe/blob/dev/src/less/parse.js#L281
Imagine you are a parser and you meet a string that starts with `(`. If you want to check if this substring is `arguments` node what really happens is you check whether everything you meet until matching `)` looks like a single argument.
What you need to change is tell parser that `parentheses` is a valid argument.
You can make this change somewhere here: https://github.com/tonyganch/gonzales-pe/blob/dev/src/less/parse.js#L297
I think parentheses should not conflict with other arguments so you can add the line at the end.
And here you add `get` function for parentheses: https://github.com/tonyganch/gonzales-pe/blob/dev/src/less/parse.js#L325

After that you can remove your changes inside `getArguments` and it should do the trick.

**Comment by tonyganch** [comment]
> what's the difference between the dev and master branch? Which one should I be PR'ing?

`master` builds parsing tree as arrays while in `dev` we build objects.
Here are some examples from `master`: https://github.com/tonyganch/gonzales-pe/blob/master/doc/AST-Description.md
I think that objects are more convenient to use and do some work to rewrite the parser to use them instead. The reason `dev` branch is not considered stable yet are a few planned changes in api. I need to finish them first and only then merge everything to `master`.

You've chosen the right branch and I'm very happy about it, thank you :apple:

**Comment by aquach** [comment]
Thanks so much for the patient explanation! I've made the changes you described.

**Comment by tonyganch** [comment]
@aquach, merged into `dev`. Thank you very much!

**Comment by aquach** [comment]
:+1::+1:
